spec/Request/users_controller_spec.rb

```diff
@@ -0,0 +1,69 @@
+require 'rails_helper'
+require 'rails-controller-testing'
+
+RSpec.describe UsersController, type: :controller do
+  describe 'GET #index' do
+    it 'returns a success response' do
+      # Perform the request
+      get :index
+
+      # Expect the response status to be 200 (OK)
+      expect(response).to have_http_status(200)
+    end
+
+    it 'renders the index template' do
+      # Perform the request
+      get :index
+
+      # Expect the correct template to be rendered
+      expect(response).to render_template(:index)
+    end
+
+    it 'assigns all users to @users' do
+      # Create some sample users
+      user1 = User.create(name: 'Nikita')
+      user2 = User.create(name: 'Janelle')
+
+      # Perform the request
+      get :index
+
+      # Expect the @users instance variable to contain all users
+      expect(assigns(:users)).to match_array([user1, user2])
+    end
+  end
+
+  describe 'GET #show' do
+    it 'returns a success response' do
+      # Create a sample user
+      User.create(name: 'Nikita')
+
+      # Perform the request
+      get :show, params: { id: 1 }
+
+      # Expect the response status to be 200 (OK)
+      expect(response).to have_http_status(200)
+    end
+
+    it 'renders the show template' do
+      # Create a sample user
+      user = User.create(name: 'Tom')
+
+      # Perform the request
+      get :show, params: { id: user.id }
+
+      # Expect the correct template to be rendered
+      expect(response).to render_template(:show)
+    end
+
+    it 'assigns the correct user to @user' do
+      # Create a sample user
+      user = User.create(name: 'Tom')
+
+      # Perform the request
+      get :show, params: { id: user.id }
+
+      # Expect the @user instance variable to contain the correct user
+      expect(assigns(:user)).to eq(user)
+    end
```

Comment:
👏 Great job on your progress! you checked the @user instance variable.  However, there's a minor adjustment needed:

- [ ] **Placeholder Test:** As per [the project requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/controller_specs_project.md#:~:text=If%20the%20response%20body%20includes%20correct%20placeholder%20text.), it's recommended to add a test case that checks for placeholders or specific content in your views. For instance, if you have a `<h1>User Details</h1>` title on your show page, you can create a test to ensure that this placeholder or specific content is present.
- [ ] Please follow the project requirement and add  testcases for users#index, users#show, posts#index, posts#show page. 🙏 

By incorporating this test, you'll enhance the completeness of your test suite and ensure that critical content is displayed as expected. Keep up the good work! 🚀👍